core/primitives/src/block_header.rs

```diff
@@ -760,7 +760,8 @@ impl BlockHeader {
             block_merkle_root,
         };
 
-        if ProtocolFeature::ChunkEndorsementsInBlockHeader.enabled(this_epoch_protocol_version) {
+        #[allow(deprecated)]
+        if ProtocolFeature::BlockHeaderV4.enabled(this_epoch_protocol_version) {
```

Comment:
Assume `ProtocolFeature::BlockHeaderV4.enabled(this_epoch_protocol_version)` is always true and simplify code. Remove `#[allow(deprecated)]`